server/webapp/WEB-INF/rails.new/webpack/single_page_apps/preferences.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+;(function () { // eslint-disable-line no-extra-semi
+  "use strict";
+
+  const m = require("mithril");
+  const Stream = require("mithril/stream");
+
+  const EmailSettingsView = require('views/preferences/email_settings');
+  const FiltersListView   = require('views/preferences/filters_list');
+  const AddFilterView     = require('views/preferences/add_filter');
+
+  const EmailSettingsModel = require('models/preferences/email_settings_model');
+  const FiltersModel       = require('models/preferences/filters_model');
+  const PipelinesModel     = require('models/preferences/pipelines_model');
+
+  document.addEventListener("DOMContentLoaded", () => {
+
+    function dataAttr(node, name) {
+      return node.getAttribute(`data-${name}`);
+    }
+
+    const main        = document.getElementById("notification-prefs");
+    const validations = document.getElementById("validations");
+    const pipelines   = JSON.parse(dataAttr(main, "pipelines"));
+
+    const userUrl     = dataAttr(main, "user-url");
+    const filtersUrl  = dataAttr(main, "filters-url");
+    const smtpEnabled = JSON.parse(dataAttr(main, "smtp-configured"));
+
+    const errorsModel = Stream();
+    const emailSettingsModel = new EmailSettingsModel(userUrl, errorsModel);
+    const filtersModel       = new FiltersModel(filtersUrl, errorsModel);
+    const pipelinesModel     = new PipelinesModel(pipelines);
+
+    const EmailSettings = new EmailSettingsView(emailSettingsModel, smtpEnabled);
+    const AddFilter     = new AddFilterView(filtersModel, pipelinesModel);
+    const Filters       = new FiltersListView(filtersModel);
+
+    const ErrorMessage = {
+      oninit(vnode) {
+        vnode.state.errors = vnode.attrs.errors;
+      },
+
+      view(vnode) {
+        if (vnode.state.errors()) {
+          return m("div", {class: "error"}, m("i", {class: "fa fa-exclamation-circle"}), vnode.state.errors());
```

Comment:
The `.callout.alert` is styled very differently than the current errors on the original preferences page. Is that alright?

For reference, this is the current style:
![image](https://cloud.githubusercontent.com/assets/723642/24779157/5a59fb3c-1ae3-11e7-9dca-d3eea7db6c05.png)

Here is `.callout.alert`:
![image](https://cloud.githubusercontent.com/assets/723642/24779181/77bf5fe6-1ae3-11e7-85fe-e6fea5fe61d7.png)